linodecli/cli.py

```diff
@@ -479,6 +482,59 @@ def _flatten_url_path(tag):
         new_tag = re.sub(r"[^a-z ]", "", new_tag).replace(" ", "-")
         return new_tag
 
+    @staticmethod
+    def merge_results_data(results: Iterable[dict]):
+        """Merge multiple json response into one"""
+        iterator = iter(results)
+        merged_result = next(iterator, None)
+        if not merged_result:
+            return None
+
+        if "pages" in merged_result:
+            merged_result["pages"] = 1
+        if "page" in merged_result:
+            merged_result["page"] = 1
+        if "data" in merged_result:
+            merged_result["data"] += list(
+                itertools.chain.from_iterable(r["data"] for r in iterator)
+            )
+        return merged_result
+
+    def _get_all_pages_results_generator(
+        self,
+        operation: CLIOperation,
+        args: List[str],
+        pages_needed: Iterable[int],
+    ):
+        for p in pages_needed:
+            self.page = p
+            yield do_request(self, operation, args).json()
+
+    def get_all_pages(self, operation: CLIOperation, args: List[str]):
+        """
+        Receive all pages of a resource from multiple
+        API responses then merge into one page.
+        """
+        print(operation, args)
+        self.page_size = 500
+        self.page = 1
+        result = do_request(self, operation, args).json()
+
+        total_pages = result.get("pages")
+
+        if total_pages and total_pages > 1:
+            pages_needed = range(2, total_pages + 1)
+
+            result = self.merge_results_data(
+                itertools.chain(
+                    (result,),
+                    self._get_all_pages_results_generator(
+                        operation, args, pages_needed
+                    ),
+                )
+            )
+        return result
```

Comment:
This looks great! Could we add some unit tests to make sure all of these function work as expected with some mock data?

Reply:
Thank you! Test added.